cranelift/frontend/src/switch.rs

```diff
@@ -289,8 +289,11 @@ impl Switch {
 }
 
 fn icmp_imm_u128(bx: &mut FunctionBuilder, cond: IntCC, x: Value, y: u128) -> Value {
-    if let Ok(index) = u64::try_from(y) {
-        bx.ins().icmp_imm(cond, x, index as i64)
+    if bx.func.dfg.value_type(x) != types::I128 {
+        assert!(u64::try_from(y).is_ok());
+        bx.ins().icmp_imm(cond, x, y as i64)
```

Comment:
This assert/conversion combination is a little surprising. If `y` converts to `u64`, doesn't that mean that there's a possibility that it would produce a negative `i64` value?

Reply:
A negative i64 here is fine as the value on which we are operating is at most a 64bit integer and thus no implicit sign extension of the immediate will be done by `icmp_imm`.